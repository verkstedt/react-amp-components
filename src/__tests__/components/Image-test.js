import React from 'react'

import Image from '../../components/Image'
import { renderComponent, renderToStaticMarkup } from '../test-utils'

describe('Image', () => {
  it('matches the snapshot', () => {
    const image = renderComponent(<Image src="https://placekitten.com/200/300" />)
    expect(image.toJSON()).toMatchSnapshot()
  })

  it('passes down the right props', () => {
    const props = {
      src: 'https://placekitten.com/200/300',
      srcset: '/img/amp.jpg 1080w, /img/amp-900.jpg 900w',
      alt: 'just an alt',
      layout: 'responsive',
      width: '100',
      height: '100',
      class: 'such-a-class'
    }
    const image = renderComponent(<Image {...props} />)
    expect(image.toJSON().props).toMatchObject(props)
  })

  it('renders only the right props', () => {
    const validProps = {
      src: 'https://placekitten.com/200/300',
      srcset: '/img/amp.jpg 1080w, /img/amp-900.jpg 900w',
      alt: 'just an alt',
      layout: 'responsive',
      width: '100',
      height: '100',
      class: 'such-a-class'
    }
    const invalidProps = {
      randomStuff: 'aiosdo',
      otherStuff: 123
    }
    const image = renderComponent(
      <Image {...{ ...validProps, ...invalidProps }} />
    )
    expect(image.toJSON().props).toMatchObject(validProps)

    Object.keys(invalidProps).forEach(prop =>
      expect(image.toJSON().props).not.toHaveProperty(prop)
    )
  })
})

