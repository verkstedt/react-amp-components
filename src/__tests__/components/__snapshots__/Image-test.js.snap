// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image matches the snapshot 1`] = `
<amp-img
  alt=""
  src="https://placekitten.com/200/300"
  srcset={null}
/>
`;
